### PR TITLE
SR-10611: ProcessInfo.activeProcessorCount too high in containerised Linux

### PR DESCRIPTION
**Pull request by spevans**
- On Linux, use sched_getaffinity() falling back to
  sysconf(_SC_NPROCESSORS_ONLN) as this should provide a better
  value for ProcessInfo.activeProcessorCount when running under
  cgroups (eg Docker).

Not sure if this is the best solution and will need an offline test in a Docker environment.

### CODE REVIEW

**Comment by spevans** [comment]
cc @ianpartridge Does this seem reasonable when running under Docker? The solutions used in other languages mentioned in https://bugs.swift.org/browse/SR-10611?filter=10905 seem to use this method.

**Comment by spevans** [comment]
@swift-ci test

**Comment by millenomi** [comment]
@swift-ci please test

File: CoreFoundation/Base.subproj/CFUtilities.c [hunk 1]
```diff
@@ -451,6 +454,14 @@ CF_PRIVATE CFIndex __CFActiveProcessorCount() {
         pcnt = 0;
     }
 #elif DEPLOYMENT_TARGET_LINUX
+
+#ifdef _SCHED_H
```
Comment (compnerd) [inline CoreFoundation/Base.subproj/CFUtilities.c]: Please use cmake for this:

```
include(CheckSymbolExists)
include(CheckIncludeFile)
check_include_file("sched.h" HAVE_SCHED_H)
check_symbol_exists(sched_getaffinity "sched.h" HAVE_SCHED_GETAFFINITY)
```

**Comment by spevans** [comment]
@swift-ci test

**Comment by spevans** [comment]
@swift-ci test

File: CoreFoundation/CMakeLists.txt [hunk 2]
```diff
@@ -337,6 +337,11 @@ if(CMAKE_SYSTEM_NAME STREQUAL Linux OR CMAKE_SYSTEM_NAME STREQUAL Android)
                              PRIVATE
                                -DDEPLOYMENT_TARGET_LINUX
                                -D_GNU_SOURCE)
+  if(HAVE_SCHED_GETAFFINITY)
+    target_compile_definitions(CoreFoundation
+                               PRIVATE
+                               -DHAVE_SCHED_GETAFFINITY)
```
Comment (compnerd) [inline CoreFoundation/CMakeLists.txt]: A level of indentation would be nice.

**Comment by spevans** [comment]
@swift-ci test and merge

**Comment by spevans** [comment]
@swift-ci test and merge

**Pull request merged**